### PR TITLE
TST: Refactor backend tests

### PR DESCRIPTION
**Pull request by datapythonista**
Reopening #2566 again.

What I did was to instead of:
```python
from ibis.backends.sqlite.tests.conftest import SQLiteTest

@pytest.mark.only_on_backends([SQLiteTest])
def test_whatever():
    pass
```

Use:
```python
@pytest.mark.only_on_backends(['sqlite'])
def test_whatever():
    pass
```
And adapted the code that handles that, also cleaning and simplifying it, since it wasn't really great code.

### CODE REVIEW

**Comment by jreback** [comment]
@datapythonista pls lmk when ready with this

**Comment by datapythonista** [comment]
I've been researching the error in the tests. Looks like the `xpass_backends` is not working properly in master, and it's being fixed here.

For example, `ibis/tests/all/test_temporal.py::test_to_timestamp` with `unit='us'` passes in master for the dask backend, but since `dask` it's not in the `xpass_backends` list it should be failing, but it doesn't.

I'm going to add the missing backends in the `xpass_backends` lists here, so this pass the tests. And then in a follow up I think I'll replace the `xpass_backends` by `xfail_backends`. It will be a bit more verbose, but it'll simplify things (including being able to test backends that are unknown to ibis, like mssql, arrow and in the future omnisci)

**Comment by jreback** [comment]
what happens if there is an invalid name in the skip list? e.g. 'dasky' do we get an error?

**Comment by datapythonista** [comment]
> what happens if there is an invalid name in the skip list? e.g. 'dasky' do we get an error?

An error should be raise, yes. Not sure if I'll have to temporary remove that error, since for a while I want to remove the omniscidb backend after this, but I want to leave the xfails for omnisci in the tests. I'll see exactly what can be done.

File: ci/run_tests.sh [hunk 1]
```diff
@@ -3,7 +3,7 @@
 # - PYTEST_BACKENDS: Space-separated list of backends to run
 # - PYTEST_EXPRESSION: Marker expression, for example "not udf"
 
-TESTS_DIRS="ibis/tests"
+TESTS_DIRS="ibis/tests ibis/backends/tests ibis/backends/base*/tests"
```
Comment (gerrymanoim) [inline ci/run_tests.sh:6]: I don't think this is actually picking up the tests in `ibis/backends/base*/tests`. If I look at tests running in https://github.com/ibis-project/ibis/pull/2576/checks?check_run_id=1651224874, I don't see tests from `ibis/backends/base_file/tests/test_schema.py` running (that seems to be the only `base*` that has `tests`. 

Were the files in https://github.com/datapythonista/ibis/tree/refactor_backend_tests/ibis/tests/sql also supposed to move to these `base_` dirs?

Comment (datapythonista) [reply ci/run_tests.sh:6]: I do see those tests running: https://github.com/ibis-project/ibis/pull/2576/checks?check_run_id=1651224874#step:4:98

> Were the files in https://github.com/datapythonista/ibis/tree/refactor_backend_tests/ibis/tests/sql also supposed to move to these base_ dirs?

Yes, I think so. There should be an issue for it if I'm not wrong.

Comment (gerrymanoim) [reply ci/run_tests.sh:6]: Oh - weird, not sure where I was looking that I missed that!

**Comment by jreback** [comment]
> looking for now-outdated files... /ibis/docs/source/backends/omnisci.rst.rst:50: WARNING: autosummary: failed to import OmniSciDBClient.sql


failed the doc build i think

**Comment by datapythonista** [comment]
Yes, I don't understand why, I checked locally and had a look at the logs but couldn't find the problem, seems unrelated to the changes here. I'll continue checking when I have an idea on what could be the problem.

**Comment by jreback** [comment]
> Yes, I don't understand why, I checked locally and had a look at the logs but couldn't find the problem, seems unrelated to the changes here. I'll continue checking when I have an idea on what could be the problem.

kk great. happy to merge this now ? or after you give a debug, up to yo.

**Comment by datapythonista** [comment]
If you are ok to merge this now, the next PR I'll finish is the one deleting omnisci, so that will fix the problem on the docs not finding omnisci, and the CI shouldn't be affected, even if this is causing the problem (which I don't see how it can be).

**Review by jreback — APPROVED** [review]

**Pull request merged**